src/cascadia/TerminalApp/CascadiaSettingsSerialization.cpp

```diff
@@ -103,145 +102,161 @@ static void _CatchRethrowSerializationExceptionWithLocationInfo(std::string_view
+
+            // Compare to the defaults.json one that we set on install.
+            // If it's different, log what the user chose.
+            if (hardcodedDefaultGuid != guid)
+            {
+                TraceLoggingWrite(
+                    g_hTerminalAppProvider, // handle to TerminalApp tracelogging provider
+                    "CustomDefaultProfile",
+                    TraceLoggingDescription("Event emitted when user has chosen a different default profile than hardcoded one on load/reload"),
+                    TraceLoggingGuid(guid, "DefaultProfile", "ID of user-chosen default profile"),
+                    TraceLoggingKeyword(MICROSOFT_KEYWORD_MEASURES),
+                    TelemetryPrivacyDataTag(PDT_ProductAndServiceUsage));
+            }
+
+            // If the user had keybinding settings preferences, we want to learn from them to make better defaults
+            auto userKeybindings = resultPtr->_userSettings[JsonKey(KeybindingsKey)];
+            if (!userKeybindings.empty())
+            {
+                // If there are custom key bindings, let's understand what they are because maybe the defaults aren't good enough
+
+                // Run it through the object so we can parse it apart and then only serialize the fields we're interested in
+                // and avoid extraneous data.
+                auto akb = winrt::make_self<AppKeyBindings>();
+                akb->LayerJson(userKeybindings);
+                auto value = akb->ToJson();
+
+                // Reserialize the keybindings
+                Json::StreamWriterBuilder wbuilder;
+                // Use 4 spaces to indent instead of \t
+                wbuilder.settings_["indentation"] = "    ";
+                wbuilder.settings_["enableYAMLCompatibility"] = true; // suppress spaces around colons
+
+                const auto keybindingsString = Json::writeString(wbuilder, value);
+
+                TraceLoggingWrite(
+                    g_hTerminalAppProvider, // handle to TerminalApp tracelogging provider
+                    "CustomKeybindings",
+                    TraceLoggingDescription("Event emitted when custom keybindings are identified on load/reload"),
+                    TraceLoggingUtf8String(keybindingsString.c_str(), "Keybindings", "Keybindings as JSON"),
+                    TraceLoggingKeyword(MICROSOFT_KEYWORD_MEASURES),
+                    TelemetryPrivacyDataTag(PDT_ProductAndServiceUsage));
+            }
         }
-    }
 
-    return *resultPtr;
+        return *resultPtr;
+    }
+    catch (const SettingsException& ex)
+    {
+        auto settings = winrt::get_self<implementation::CascadiaSettings>(LoadDefaults());
+        settings->_loadError = ex.Error();
+        return *settings;
+    }
+    catch (const SettingsTypedDeserializationException& e)
+    {
+        auto settings = winrt::get_self<implementation::CascadiaSettings>(LoadDefaults());
+        std::string_view what{ e.what() };

```

Comment:
i'm not in love with the number of copies of this "fall back to the defaults" code there are now. this is interesting ...
it's a process that can fail, and we need a way to claw an object back out of it just to hang errors off of it.

when the consumer gets an error, what is it going to do? call LoadDefaults _again_?

Reply:
Yeah. I could probably get away with just returning a blank `CascadiaSettings` instead? Especially since the first thing the consumer does is "if loading the settings failed --> load defaults --> display error/warning message"

This is all _absolutely awful_.